torch/backends/cudnn/__init__.py

```diff
@@ -376,9 +378,25 @@ def descriptor(tensor, N=None):
         tensor = tensor.view(tensor.size(0), tensor.size(1), 1, 1)
     elif tensor.dim() == 3:
         tensor = tensor.view(tensor.size(0), tensor.size(1), tensor.size(2), 1)
-    descriptor.set(tensor)
+    if N is not None:
+        descriptor.set_all(tensor)
+    else:
+        descriptor.set(tensor)
     return descriptor
 
+
+def descriptor_sequence(tensor, batch_sizes):
```

Comment:
I used batch_sizes, because that's the name I used everywhere else. I don't think this function will be useful for anything else than variable sequences. I think it makes it clear, because sizes sound a bit like lengths, while it's really a different representation (there are B lengths, but max(T) sizes).

Sure, I'll make it consistent. Do you want me to rewrite `set_all` to accept these tuples instead of a tensor?